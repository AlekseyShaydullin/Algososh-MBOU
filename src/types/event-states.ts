export enum EventStates {
  NoEvent = 'noEvent',
  AddToHead = 'addToHead',
  AddToTail = 'addToTail',
  DeleteFromHead = 'deleteFromHead',
  DeleteFromTail = 'deleteFromTail',
  AddToIndex = 'addToIndex',
  DeleteByIndex = 'deleteByIndex',
}