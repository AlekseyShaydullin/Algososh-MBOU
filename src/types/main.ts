import { ElementStates } from "./element-states";

export type TArrayString = {
  letter?: string;
  state: ElementStates
};
