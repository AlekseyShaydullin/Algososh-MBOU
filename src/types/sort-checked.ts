export enum sortChecked {
  selection = "selection",
  bubble = "bubble",
}