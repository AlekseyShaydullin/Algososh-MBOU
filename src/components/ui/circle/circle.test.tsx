import { Circle } from "./circle";
import TestRenderer from 'react-test-renderer';
import { ElementStates } from "../../../types/element-states";

describe('Circle component test', () => {

  it('Circle without a letter', () => {
    const circle = TestRenderer.create(<Circle />).toJSON();
    expect(circle).toMatchSnapshot()
  });

  it('Circle with letters', () => {
    const circle = TestRenderer.create(<Circle letter={'H'} />).toJSON();
    expect(circle).toMatchSnapshot()
  });
  
  it('Circle with head', () => {
    const circle = TestRenderer.create(<Circle head={'0'} />).toJSON();
    expect(circle).toMatchSnapshot()
  });

  it('Circle with a react element in head', () => {
    const circle = TestRenderer.create(<Circle head={<Circle/>} />).toJSON();
    expect(circle).toMatchSnapshot()
  });

  it('Circle with tail', () => {
    const circle = TestRenderer.create(<Circle tail={'0'} />).toJSON();
    expect(circle).toMatchSnapshot()
  });

  it('Circle with a react element in tail', () => {
    const circle = TestRenderer.create(<Circle tail={<Circle/>} />).toJSON();
    expect(circle).toMatchSnapshot()
  });

  it('Circle with index', () => {
    const circle = TestRenderer.create(<Circle index={0} />).toJSON();
    expect(circle).toMatchSnapshot()
  });

  it('Circle with props isSmall', () => {
    const circle = TestRenderer.create(<Circle isSmall />).toJSON();
    expect(circle).toMatchSnapshot()
  });

  it('Circle with Default state works correctly', () => {
    const circle = TestRenderer.create(<Circle state={ElementStates.Default} />).toJSON();
    expect(circle).toMatchSnapshot()
  });

  it('Circle with Changing state works correctly', () => {
    const circle = TestRenderer.create(<Circle state={ElementStates.Changing} />).toJSON();
    expect(circle).toMatchSnapshot()
  });

  it('Circle with Modified state works correctly', () => {
    const circle = TestRenderer.create(<Circle state={ElementStates.Modified} />).toJSON();
    expect(circle).toMatchSnapshot()
  });
})