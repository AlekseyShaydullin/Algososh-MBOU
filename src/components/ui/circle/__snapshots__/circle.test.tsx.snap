// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component test Circle with Changing state works correctly 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component test Circle with Default state works correctly 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component test Circle with Modified state works correctly 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component test Circle with a react element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component test Circle with a react element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component test Circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    0
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component test Circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle component test Circle with letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      H
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component test Circle with props isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component test Circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    0
  </div>
</div>
`;

exports[`Circle component test Circle without a letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
